SmartDeviceLinkTests/SDLCacheFileManagerSpec.m

```diff
@@ -0,0 +1,313 @@
+    __block NSString *testURL = nil;
+    __block SDLOnSystemRequest *expiredTestRequest = nil;
+    __block NSString *testFilePath = nil;
+    __block NSString *expiredTestURL = nil;
+    __block UIImage *testImage = nil;
+    __block SDLIconArchiveFile *testArchiveFile = nil;
+    __block SDLLockScreenIconCache *testIconCache = nil;
+    __block SDLLockScreenIconCache *testExpiredIconCache = nil;
+    __block NSArray<SDLLockScreenIconCache *> *testArchiveFileLockScreenCacheArray = nil;
+
+    beforeEach(^{
+        testManager = [[SDLCacheFileManager alloc] init];
+        testManagerMock = OCMPartialMock(testManager);
+
+        testFilePath = [[NSBundle bundleForClass:self.class] pathForResource:@"testImagePNG" ofType:@"png"];
+        testImage = [UIImage imageNamed:@"testImagePNG" inBundle:[NSBundle bundleForClass:self.class] compatibleWithTraitCollection:nil];
+
+        mockFileManager = OCMStrictClassMock([NSFileManager class]);
+        testManager.fileManager = mockFileManager;
+        mockUnarchiver = OCMClassMock([NSKeyedUnarchiver class]);
+
+        mockUIImage = OCMClassMock([UIImage class]);
+        mockDataTask = OCMClassMock([NSURLSession class]);
+
+        mockArchiver = OCMClassMock([NSKeyedArchiver class]);
+    });
+
+    it(@"should initialize properties", ^{
+        expect(testManager.urlSession).toNot(beNil());
+        expect(testManager.fileManager).toNot(beNil());
+    });
+
+    describe(@"request with existing icon", ^{
+        beforeEach(^{
+            testURL = @"testURL";
+            testRequest = [[SDLOnSystemRequest alloc] init];
+            testRequest.url = testURL;
+            testIconCache = [[SDLLockScreenIconCache alloc] initWithIconUrl:testURL iconFilePath:testFilePath];
+
+            expiredTestURL = @"expiredTestURL";
+            expiredTestRequest = [[SDLOnSystemRequest alloc] init];
+            expiredTestRequest.url = expiredTestURL;
+            testExpiredIconCache = [[SDLLockScreenIconCache alloc] initWithIconUrl:expiredTestURL iconFilePath:testFilePath];
+            testExpiredIconCache.lastModifiedDate = [[NSDate date] dateByAddingTimeInterval:-31*24*60*60];
+
+            testArchiveFileLockScreenCacheArray = @[testIconCache,testExpiredIconCache];
+
+            testArchiveFile = [[SDLIconArchiveFile alloc] init];
+            testArchiveFile.lockScreenIconCaches = testArchiveFileLockScreenCacheArray;
+
+            OCMStub(ClassMethod([mockUnarchiver unarchiveObjectWithFile:[OCMArg any]])).andReturn(testArchiveFile);
+            OCMStub([mockFileManager fileExistsAtPath:[OCMArg any]]).andReturn(YES);
+        });
+
+        context(@"icon is not expired", ^{
+            __block UIImage *resultImage = nil;
+            __block NSError *resultError = nil;
+
+            context(@"failure to retrieve icon", ^{
+                beforeEach(^{
+                    OCMStub(ClassMethod([mockArchiver archiveRootObject:[OCMArg any] toFile:[OCMArg any]])).andReturn(YES);
```

Comment:
Instead of `OCMArg any`, you can do `OCMArg isNotNil` to require that it be non-nil at this point. Take a look throughout this file and check for places you can require it to not be nil.